File: 2023/01 - January/26/03/Artwork.java
// Artwork.java

// Represents An Artwork
public class Artwork {
	// Declaring Required Variables
	private String title;
	private int yearCreated;
	private Artist artist;

	// Default Constructor
	public Artwork() {
		title = "unknown";
		yearCreated = -1;
		artist = new Artist();
	}

	// Getters For Title
	public String getTitle() {
		return title;
	}

	// Getters For Year Created
	public int getYearCreated() {
		return yearCreated;
	}

	// Parameterized Constructor
	public Artwork(String title, int yearCreated, Artist artist) {
		this.title = title;
		this.yearCreated = yearCreated;
		this.artist = artist;
	}

	// Printing Information
	public void printInfo() {
		// Call Artist Print Method
		artist.printInfo();
		System.out.println("Title: " + title + ", " + yearCreated);
	}
}

File: 2023/02 - February/06/04/Artwork.cpp
#include "Artwork.h"

// Default Constructor
Artwork::Artwork() : Artwork("unknown", -1, Artist()) { }

// Parameterized Constructor
Artwork::Artwork(std::string title, int yearCreated, Artist artist) {
	this->title = title;
	this->yearCreated = yearCreated;
	this->artist = artist;
}

// Getter For Title
std::string Artwork::GetTitle() const {
	return title;
}


// Getter For Year Created
int Artwork::GetYearCreated() const {
	return yearCreated;
}

// Prints Artwork Info
void Artwork::PrintInfo() const {
	// Print Artist Info
	artist.PrintInfo();
	// Print Title And Year Created
	std::cout << "Title: " << title << ", " << yearCreated << std::endl;
}
